service/controller/chart/v1/resource/status/create.go

```diff
@@ -70,6 +67,9 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		if key.IsCordoned(cr) {
 			status = releaseStatusCordoned
 			reason = key.CordonReason(cr)
+		} else if cc.Status.Reason != "" {
+			// If a reason was added to the controller context something went wrong.
+			reason = cc.Status.Reason
```

Comment:
Usually this is problems pulling chart tarballs like 404s. Since helm doesn't know about this.

Reply:
This background knowledge should better be added to the comment. When I read the sentence above in the comment it does not tell me much somehow.

Reply:
Yes I'll add more detail.